common/client/node_lifecycle.go

```diff
@@ -423,18 +420,78 @@ func (n *node[CHAIN_ID, HEAD, RPC]) invalidChainIDLoop() {
 		case <-n.nodeCtx.Done():
 			return
 		case <-time.After(chainIDRecheckBackoff.Duration()):
-			err := n.verify(n.nodeCtx)
-			if errors.Is(err, errInvalidChainID) {
-				lggr.Errorw("Failed to verify RPC node; remote endpoint returned the wrong chain ID", "err", err)
+			state := n.verifyConn(n.nodeCtx, lggr)
+			switch state {
+			case nodeStateInvalidChainID:
 				continue
-			} else if err != nil {
-				lggr.Errorw(fmt.Sprintf("Unexpected error while verifying RPC node chain ID; %v", err), "err", err)
+			case nodeStateAlive:
+				lggr.Infow(fmt.Sprintf("Successfully verified RPC node. Node was offline for %s", time.Since(invalidAt)), "nodeState", n.State())
+				fallthrough
+			default:
+				n.declareState(state)
+				return
+			}
+		}
+	}
+}
+
+func (n *node[CHAIN_ID, HEAD, RPC]) syncingLoop() {
+	defer n.wg.Done()
+
+	{
+		// sanity check
+		state := n.State()
+		switch state {
+		case nodeStateSyncing:
+		case nodeStateClosed:
+			return
+		default:
+			panic(fmt.Sprintf("syncingLoop can only run for node in nodeStateSyncing state, got: %s", state))
+		}
+	}
+	syncingAt := time.Now()
+
+	lggr := logger.Sugared(logger.Named(n.lfcLog, "Syncing"))
+	lggr.Debugw(fmt.Sprintf("Periodically re-checking RPC node %s with syncing status", n.String()), "nodeState", n.State())
+	// Need to redial since syncing nodes are automatically disconnected
+	state := n.createVerifiedConn(n.nodeCtx, lggr)
+	if state != nodeStateSyncing {
+		n.declareState(state)
+		return
+	}
+
+	recheckBackoff := iutils.NewRedialBackoff()
+
+	for {
+		select {
+		case <-n.nodeCtx.Done():
+			return
+		case <-time.After(recheckBackoff.Duration()):
+			lggr.Tracew("Trying to recheck if the node is still syncing", "nodeState", n.State())
+			isSyncing, err := n.rpc.IsSyncing(n.nodeCtx)
+			if err != nil {
+				lggr.Errorw("Unexpected error while verifying RPC node synchronization status", "err", err, "nodeState", n.State())
 				n.declareUnreachable()
 				return
 			}
-			lggr.Infow(fmt.Sprintf("Successfully verified RPC node. Node was offline for %s", time.Since(invalidAt)), "nodeState", n.State())
+
+			if isSyncing {
+				lggr.Errorw("Verification failed: Node is syncing", "nodeState", n.State())
+				continue
+			}
+
+			lggr.Infow(fmt.Sprintf("Successfully verified RPC node. Node was syncing for %s", time.Since(syncingAt)), "nodeState", n.State())
 			n.declareAlive()
 			return
+		case <-time.After(zombieNodeCheckInterval(n.noNewHeadsThreshold)):
+			if n.nLiveNodes != nil {
+				if l, _, _ := n.nLiveNodes(); l < 1 {
+					lggr.Critical("RPC endpoint is still syncing, but there are no other available nodes. This RPC node will be forcibly moved back into the live pool in a degraded state")
```

Comment:
Is this helpful?
Do we know if a syncing node can operate even in a degraded state?
For example, if it is not caught upto latest heads, then it can't serve heads, or check for receipts, or verify transactions sent to it.
Isn't it better to just let it sync first?